overlord/servicestate/helpers.go

```diff
@@ -97,3 +98,27 @@ func updateSnapstateServices(snapst *snapstate.SnapState, enable, disable []*sna
 	}
 	return true, nil
 }
+
+// MakeServiceQuotaMap builds a map of services and their quota groups. The goal is
+// to make sure that if any quota group applies to a snap service, then we can easily
+// look it up in this based on it's snap.AppInfo. We take the snap instance name, the services
+// want to include in the map (should only be services of the snap name provided) and the quota
+// group of the snap.
+func MakeServiceQuotaMap(snapInfo *snap.Info, grp *quota.Group) map[*snap.AppInfo]*quota.Group {
+	snapServices := snapInfo.Services()
+	if len(snapServices) == 0 || grp == nil {
+		return nil
+	}
+
+	svcQuotaMap := make(map[*snap.AppInfo]*quota.Group)
+	for _, svc := range snapServices {
+		// always default to the snap quota group if the service is not
+		// in a separate one
+		svcGrp := grp.GroupForService(fmt.Sprintf("%s.%s", snapInfo.InstanceName(), svc.Name))
+		if svcGrp == nil {
+			svcGrp = grp
+		}
```

Comment:
because of my other comment I still feel leaving this as nil should be fine, but this is also ok